Make examples commands return a failure

adbridge opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->
Currently the following commands in examples.py,
do_import()
do_deploy()
do_versionning()
do_clone()

all return a success status (ie 0) irrespective of any errors
originating from their sub-functions.

This PR fixes this. Now these commands will return one of:
0 - success
1 - general failure
x - failure returned by a subprocess.call function



### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

adbridge committed:
Make examples commands return a failure

Currently the following commands in examples.py,
do_import()
do_deploy()
do_versionning()
do_clone()

all return a success status (ie 0) irrespective of any errors
originating from their sub-functions.

This PR fixes this. Now these commands will return one of:
0 - success
1 - general failure
x - failure returned by a subprocess.call function

ciarmcom commented:
@adbridge, thank you for your changes. @ARMmbed/mbed-os-tools @ARMmbed/mbed-os-maintainers please review.

adbridge commented:
Fixes #8568

OPpuolitaival commented:
@adbridge Did you test this? I tested a bit suprocess.call() function and seems to work in a way that:
- subprocess.call("exit 1") => not returning anything, just raising an error
- subprocess.call("exit 1", shell=True) => return a return code

It seems that you expect the behaviour as shell=True but it is false as default

OPpuolitaival commented:
[examples] Running shell script
+ python -u ../mbed-os/tools/test/examples/examples.py clone
Traceback (most recent call last):
  File "../mbed-os/tools/test/examples/examples.py", line 17, in <module>
    import examples_lib as lib
  File "/builds/ws/ci_PR-8605-QHY26DPOP4ARV3WYD2LBS/mbed-os/tools/test/examples/examples_lib.py", line 421
    return result:
                 ^
SyntaxError: invalid syntax
script returned exit code 1

OPpuolitaival commented:
Now fixed CI so that it also catch examples fetching problems right way.

cmonr commented:
> Now fixed CI so that it also catch examples fetching problems right way.

@OPpuolitaival Are you referring to this?
https://github.com/ARMmbed/mbed-os-ci/pull/190/files

bridadan commented on tools/test/examples/examples_lib.py:
```diff
@@ -405,5 +415,10 @@ def update_mbedos_version(config, tag, examples):
+            result = subprocess.call(["mbed-cli", "update", tag, "--clean"])
             os.chdir("../..")
+            if result:
+                return result:
```
@OPpuolitaival found this, but the correct thing here would be:

```suggestion
                return result
```

adbridge replied:
Yup copy paste there me thinks.

adbridge commented:
@OPpuolitaival mmmm I just looked at a call that was already in the code:

    if subprocess.call([repo_info['type'], "clone", repo_info['repo']]) == 0:

So from what you said that would never have worked either.

adbridge commented:
Fixing...

adbridge committed:
Fix up subprocess calls

subprocess.call() does not by default return a status value.

Update the commands to add shell=True which forces a return value.
Also convert the commands to a single string rather than a list as
this plays more nicely with both linux and windows.

Also fix a spurious :

NirSonnenschein commented:
/morph build
